### PR TITLE
Add geom_grid_file in Geometry class and update gridgen.yml

### LINKED ISSUE #313: Remove hardcoded file name for geometry
## Description
The soca_gridgen.x application generates a file `soca_gridspec.nc` that is subsequently read for the initialization of the Geometry class throughout soca. This currently prohibits using 2 different geometries which will be needed to resolve issue #309 .

## Solution
Pass the name of the gridspec file through the yaml config file.

## Related to
- required for #309

### PR DESCRIPTION
**Pull request by clouden90**
Added geom_grid_file in Geometry Class and update gridgen.yml(#313)

## Description
The soca_gridgen.x application generates a file soca_gridspec.nc that is subsequently read for the initialization of the Geometry class throughout soca. This currently prohibits using 2 different geometries which will be needed to resolve issue #309

### Issue(s) addressed
closes #313 

## Testing
How were these changes tested? Add grid filename in gridgen.yml 
What compilers / HPCs was it tested with? gcc
Are the changes covered by ctests? yes

## Dependencies
None

### CODE REVIEW

File: src/soca/Geometry/soca_geom_mod.F90 [hunk 1]
```diff
@@ -52,6 +52,8 @@ module soca_geom_mod
     real(kind=kind_real), allocatable, dimension(:,:) :: cell_area
     real(kind=kind_real), allocatable, dimension(:,:) :: rossby_radius
     logical :: save_local_domain = .false. ! If true, save the local geometry for each pe.
+    character(len=:), allocatable :: geom_grid_file
```
Comment (guillaumevernieres) [inline src/soca/Geometry/soca_geom_mod.F90:55]: ~Looks like it's never allocated/deallocated~ 
Ignore for now, it looks like we're doing the same thing in a bunch of other places. @travissluka , can you educate us on why it is ok? For example, [these](https://github.com/JCSDA/soca/blob/061609165c1f1d65402d70c0dd02e33bbf85af0b/src/soca/Fields/soca_fields_mod.F90#L43) are never explicitly allocated.

Comment (travissluka) [reply src/soca/Geometry/soca_geom_mod.F90:55]: https://en.wikibooks.org/wiki/Fortran/strings#CHARACTER(:),_ALLOCATABLE

fancy "new" fortran features to make life with strings easier

Comment (guillaumevernieres) [reply src/soca/Geometry/soca_geom_mod.F90:55]: :+1:

File: src/soca/Geometry/soca_geom_mod.F90 [hunk 2]
```diff
@@ -92,6 +94,13 @@ subroutine geom_init(self, f_conf)
   self%nzi = self%ice_column%nzi
   self%nzs = self%ice_column%nzs
 
+  ! User-defined grid filename
+  if ( f_conf%has("geom_grid_file") ) then
+    call f_conf%get_or_die("geom_grid_file", self%geom_grid_file)
+  else
+    self%geom_grid_file = "soca_gridspec.nc"
+  end if
+
```
Comment (travissluka) [inline src/soca/Geometry/soca_geom_mod.F90]:
```suggestion
  ! User-defined grid filename
  if ( .not. f_conf%get("geom_grid_file", self%geom_grid_file) ) then
    self%geom_grid_file = "soca_gridspec.nc" ! default if not found
  end if
```

Not at all important, but syntactically slimmer. (`get_or_die()` gets used too much in the soca code, we don't really want to die here)

File: src/soca/Geometry/soca_geom_mod.F90 [hunk 3]
```diff
@@ -92,6 +94,13 @@ subroutine geom_init(self, f_conf)
   self%nzi = self%ice_column%nzi
   self%nzs = self%ice_column%nzs
 
+  ! User-defined grid filename
+  if ( f_conf%has("geom_grid_file") ) then
+    call f_conf%get_or_die("geom_grid_file", self%geom_grid_file)
+  else
+    self%geom_grid_file = "soca_gridspec.nc"
```
Comment (guillaumevernieres) [inline src/soca/Geometry/soca_geom_mod.F90]: @clouden90 , move L101 up before the if and you can remove the else ... 1 less line :tada: :tada:

File: src/soca/Geometry/soca_geom_mod.F90 (hunk 2 above)
Comment (clouden90) [reply src/soca/Geometry/soca_geom_mod.F90]: Good Point, @travissluka . Will combine with @guillaumevernieres 's comment.

**Comment by guillaumevernieres** [comment]
@clouden90 , almost done but you are a few commits behind develop, [#317](https://github.com/JCSDA/soca/pull/317) was merged a few days ago. Safer to merge develop into your branch and push again.

**Comment by clouden90** [comment]
> @clouden90 , almost done but you are a few commits behind develop, [#317](https://github.com/JCSDA/soca/pull/317) was merged a few days ago. Safer to merge develop into your branch and push again.

Done. Plz check

**Review by guillaumevernieres — APPROVED** [review]
:+1:

**Comment by flampouris** [comment]
The very first PR of many to come @clouden90 , great job!

**Pull request merged**